Runtime/AvatarCreator/WebRequests/AvatarApi.cs

```diff
@@ -93,8 +82,9 @@ public async Task<AvatarProperties> CreateFromTemplateAvatar(string templateId,
 
             response.ThrowIfError();
 
-            var json = JObject.Parse(response.Text);
-            var data = json[DATA]!.ToString();
+            var createDraftAvatarResponse = JsonConvert.DeserializeObject<CreateDraftAvatarResponse>(response.Text);
```

Comment:
Example of point 1.

Reply:
I do personally like the idea of using Json Deserialize object as it is what I am used to and I find it to be more readable, I think it might also have some minor perfomance benefits particularly when dealing with large json payloads. However my main concern with it is that the data responses change so much with the web API's quite often a new property is release before we even hear about it (an example of that was modular asset categories some months ago). 
The other thing is that the data responses from different endpoints are not really inconsistent with their naming which might cause us to make extra classes just because they use iconUrl instead of imageUrl or something 😅 
In saying that though probably these things are more stable nowadays and probably less likely to change as time goes on.

Reply:
Another thing that kinda comes to mind as a downside of Deserialize approach is that some of our data responses have quite a nested structure which would mean we need to make more classes to hold the data

Reply:
> Another thing that kinda comes to mind as a downside of Deserialize approach is that some of our data responses have quite a nested structure which would mean we need to make more classes to hold the data

IMO this is actually a positive thing. The SDK then becomes a reference for developers to understand the types returned by the API, much like other more mature SDKs do. Like the Stripe SDK for example.

> I don't care too much about having to make classes for the responses but one thing that is annoying is that the data responses from different endpoints are really inconsistent with their naming which might cause us to make extra classes just because they use iconUrl instead of imageUrl or something

It'd be standard practice for basically every endpoint to have it's own response class, even if there is a lot of overlap between them in some endpoints. It's sort of a ball-ache at first, but then it should be fine once done. If there are any things at the API level which don't make sense then this would enable us to find them and report them, rather than just cover them up 😅

> However my main concern with it is that the data responses change so much with the web API's quite often a new property is release before we even hear about it (an example of that was modular asset categories some months ago).

JsonConvert.Deserialize can be setup to have loose mapping though, which means if properties disappear then it won't throw, it'll just not populate them, and if new properties are added then they'll just get ignored at first until the types are updated. Obviously it would be the most ideal to have the APIs be more stable, but I don't really see how using JsonConvert would be any different here to the JObject approach we have now? 🤔